ext/drx/scatter_gather_aarch64.c

```diff
@@ -594,6 +730,19 @@ unreserve_pred_register(void *drcontext, instrlist_t *bb, instr_t *where,
                            opnd_size_from_bytes(proc_get_vector_length_bytes() / 8));
 }
 
+void
+unreserve_vector_register(void *drcontext, instrlist_t *bb, instr_t *where,
+                          reg_id_t scratch_gpr0, reg_id_t scratch_vec,
+                          spill_slot_state_t *slot_state)
+{
+    DR_ASSERT(slot_state->vector_slots[0] == scratch_vec);
+    slot_state->vector_slots[0] = DR_REG_NULL;
```

Comment:
I have rewritten this code a bit and hopefully now its clearer how it is used.
`*scratch_vector_spill_slot_aligned` is now a buffer of size `NUM_VECTOR_SLOTS * vector_length_in_bytes` and `slot_state->vector_slots` is an array corresponding to the slots inside `scratch_vector_spill_slot_aligned`. It just happens that `NUM_VECTOR_SLOTS` is `1` so it only contains one slot at the moment. This is how I always envisioned it, there were just a few places before that assumed `NUM_VECTOR_SLOTS == 1` so it might not have been completely clear.